### PR TITLE
Update threads pool libraries to benchmark to latest.

### PR DESCRIPTION
**Pull request by jerome-benoit**
Signed-off-by: Jérôme Benoit <jerome.benoit@sap.com>

## PR Checklist

- [x] Please add a description of your changes.
- [ ] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [ ] If your changes contain any new feature please be sure to document it.
- [ ] Please add a link to the open issue or task that this pull request will resolve.
- [ ] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If yes please indicate it and migration info into the CHANGELOG.md file.

---

### CODE REVIEW

**Review by Shinigami92 — DISMISSED** [review]

**Comment by Shinigami92** [comment]
Wait, you should also update the benchmarks, should you?

**Comment by jerome-benoit** [comment]
> Wait, you should also update the benchmarks, should you?

It should be done by @pioardi, the documentation reference his computer. If he can't do it anymore, he needs to tell before you or I take over the benchmark runs.

**Comment by Shinigami92** [comment]
But the benchmarks should be always match to their versions
cause they can differ in results from version to version

**Comment by jerome-benoit** [comment]
> But the benchmarks should be always match to their versions
> cause they can differ in results from version to version

Theoretically, the merge of that PR shall be done after @pioardi benchmark run from that branch and the commit of the result in it.

**Review by pioardi — APPROVED** [review]

**Comment by jerome-benoit** [comment]
@pioardi you must do a benchmark run on top of that branch and commit the result.

**Comment by jerome-benoit** [comment]
> @pioardi you must do a benchmark run on top of that branch and commit the result.

@pioardi ping?

**Comment by pioardi** [comment]
Currently few available Sorry for that guys, we can Aldo jeep the doc more generico so you can run the beach Also on other machines of you want

**Comment by Shinigami92** [comment]
> Currently few available Sorry for that guys, we can Aldo jeep the doc more generico so you can run the beach Also on other machines of you want

@jerome-benoit You are really active for this repo, do you want to take your machine for that?

**Comment by jerome-benoit** [comment]
> > Currently few available Sorry for that guys, we can Aldo jeep the doc more generico so you can run the beach Also on other machines of you want
> 
> @jerome-benoit You are really active for this repo, do you want to take your machine for that?

I can do a benchmark run and remove the machine specification.

**Review by pioardi — APPROVED** [review]

File: benchmarks/versus-external-pools/BENCH-100000.md [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+| `node fixed-piscina.js` | 18.222 ± 0.519 | 17.529 | 19.164 | 1.06 ± 0.03 |
+| `node dynamic-poolifier.js` | 17.135 ± 0.256 | 16.926 | 17.755 | 1.00 |
+| `node fixed-poolifier.js` | 17.947 ± 0.489 | 17.567 | 18.951 | 1.05 ± 0.03 |
+| `node dynamic-suchmokuo-node-worker-threads-pool.js` | 25.460 ± 0.453 | 25.124 | 26.617 | 1.49 ± 0.03 |
+| `node static-suchmokuo-node-worker-threads-pool.js` | 18.993 ± 0.714 | 18.388 | 20.610 | 1.11 ± 0.04 |
+| `node threadjs.js` | 80.939 ± 3.548 | 76.881 | 88.363 | 4.72 ± 0.22 |
+| `node dynamic-workerpool.js` | 19.292 ± 0.110 | 19.109 | 19.480 | 1.13 ± 0.02 |
+| `node fixed-workerpool.js` | 19.684 ± 0.636 | 19.212 | 21.049 | 1.15 ± 0.04 |
+| `node fixed-threadwork.js` | 30.023 ± 0.362 | 29.524 | 30.727 | 1.75 ± 0.03 |
+| `node fixed-microjob.js` | 32.526 ± 0.649 | 31.696 | 33.687 | 1.90 ± 0.05 |
```
Comment (Shinigami92) [inline benchmarks/versus-external-pools/BENCH-100000.md:13]:
```suggestion
| Command                                              |       Mean [s] | Min [s] | Max [s] |    Relative |
|:-----------------------------------------------------|---------------:|--------:|--------:|------------:|
| `node dynamic-piscina.js`                            | 18.364 ± 0.203 |  18.050 |  18.737 | 1.07 ± 0.02 |
| `node fixed-piscina.js`                              | 18.222 ± 0.519 |  17.529 |  19.164 | 1.06 ± 0.03 |
| `node dynamic-poolifier.js`                          | 17.135 ± 0.256 |  16.926 |  17.755 |        1.00 |
| `node fixed-poolifier.js`                            | 17.947 ± 0.489 |  17.567 |  18.951 | 1.05 ± 0.03 |
| `node dynamic-suchmokuo-node-worker-threads-pool.js` | 25.460 ± 0.453 |  25.124 |  26.617 | 1.49 ± 0.03 |
| `node static-suchmokuo-node-worker-threads-pool.js`  | 18.993 ± 0.714 |  18.388 |  20.610 | 1.11 ± 0.04 |
| `node threadjs.js`                                   | 80.939 ± 3.548 |  76.881 |  88.363 | 4.72 ± 0.22 |
| `node dynamic-workerpool.js`                         | 19.292 ± 0.110 |  19.109 |  19.480 | 1.13 ± 0.02 |
| `node fixed-workerpool.js`                           | 19.684 ± 0.636 |  19.212 |  21.049 | 1.15 ± 0.04 |
| `node fixed-threadwork.js`                           | 30.023 ± 0.362 |  29.524 |  30.727 | 1.75 ± 0.03 |
| `node fixed-microjob.js`                             | 32.526 ± 0.649 |  31.696 |  33.687 | 1.90 ± 0.05 |
```

Comment (jerome-benoit) [reply benchmarks/versus-external-pools/BENCH-100000.md:13]: It's the default hyperfine output. It does not worth the effort to reformat it manually like I've done before, it gonna be overwritten each time.

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**